src/core/server/status/types.ts

```diff
@@ -114,6 +114,8 @@ export type ServiceStatusLevel = typeof ServiceStatusLevels[keyof typeof Service
 export interface CoreStatus {
   elasticsearch: ServiceStatus;
   savedObjects: ServiceStatus;
+  /** Summary of all core services status */
+  overall: ServiceStatus;
```

Comment:
I went back and forth on the implementation, on how to access core's overall status from the `/status` route handler, as atm, we're exposing each individual service's status, the overall (core+plugins) status, but not the 'global'/'overall' core status. 

I thought about 3 different options:

1. (PR's implementation) add a new `overall` property to `CoreStatus` (accessible via `status.core$`)

Just add a new `overall` property at the same level the core services statuses are exposed

2. Add a new `coreOverall` to `StatusServiceSetup` (would be accessible via `status.coreOverall$`)

To have a better separation between the individual services' statuses and coreOverall, we could instead add a new property to the service's contract instead.

3. Just compute the status in the `/api/status` handler

We could just use `getSummaryStatus` in `src/core/server/status/routes/status.ts` and get core's summary status from here to deduce core's overall status. 

I initially though that `1.` and `2.` were both better designs than `3` and that `1.` was going to require less changes than `2.` because we're not changing the service's root contract. However, after implementation, we can see that we're forced to manually exclude this new `overall` property in a lot of places (status computation, model used for the server-client communication...), so I'm unsure this is really the best one.

After having actually implemented the thing, I wonder if, given that currently the only identified need of this core overall status is inside the route handler, to change the response status code accordingly, `3.` wouldn't be the way to go.

What does the team think?  Do we like this implementation or should I use another approach?

Reply:
> I wonder if, given that currently the only identified need of this core overall status is inside the route handler, to change the response status code accordingly, `3.` wouldn't be the way to go.

My initial reaction is that 3 seems the simplest, and it wouldn't prevent us from switching to `1.` or `2.` later if we change our minds.

Reply:
(But let's see what others think) @elastic/kibana-core